core/lib/contracts/src/lib.rs

```diff
@@ -288,12 +292,50 @@ pub fn read_zbin_bytecode(relative_zbin_path: impl AsRef<Path>) -> Vec<u8> {
     read_zbin_bytecode_from_path(bytecode_path)
 }
 
+pub fn read_yul_bytecode(relative_artifacts_path: PathBuf, name: &str) -> Vec<u8> {
```

Comment:
Nit: Can't `relative_artifacts_path` be a `&str`? AFAICT, in all calls, the provided path *is* a string converted to `PathBuf` via `Path`.